### PR TITLE
Fix for VF device not found on 6.x

### PR DESCRIPTION
**Pull request by sireeshapotnuru**
This PR meets below requirements

1) For 6.x, VF interface is returning blank since soft link is not created.So, added a logic to get VF interface based upon the MAC address of Synthetic interface
2) Execute dhclient to get IP for the Extra NICs attached
3) Added a condition check to skip testcase for distro below 6.7, since SRIOV is supported from 6.7 onwards

### CODE REVIEW

**Comment by sireeshapotnuru** [comment]
[LISAv2 Test Results Summary]
Test Run On           : 05/02/2019 09:39:09
ARM Image Under Test  : RedHat : RHEL : 6.10 : latest
Total Test Cases      : 4 (3 Passed, 1 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:2

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION                                                    PASS                 3.79 
    2 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU                                           FAIL                 2.24 ( azure known issue - hv_netvsc: probe of 000d3a6e-1bce-000d-3a6e-1bce000d3a6e failed with error -122)
    3 SRIOV                SRIOV-VERIFY-SINGLE-VF-CONNECTION                                                 PASS                 1.43 
    4 SRIOV                SRIOV-VERIFY-SINGLE-VF-CONNECTION-MAX-VCPU                                        PASS                  1.4 



[LISAv2 Test Results Summary]
Test Run On           : 05/02/2019 10:11:19
ARM Image Under Test  : OpenLogic : CentOS : 7.5 : latest
Total Test Cases      : 4 (4 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:50

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION                                                    PASS                 4.39 
    2 SRIOV                SRIOV-VERIFY-MAX-VF-CONNECTION-MAX-VCPU                                           PASS                 3.97 
    3 SRIOV                SRIOV-VERIFY-SINGLE-VF-CONNECTION                                                 PASS                 2.13 
    4 SRIOV                SRIOV-VERIFY-SINGLE-VF-CONNECTION-MAX-VCPU                                        PASS                 1.79

**Comment by chvalean** [comment]
@adriansuhov is the SR-IOV automation maintainer, please wait for his sign-off.

File: Testscripts/Linux/SR-IOV-Utils.sh
Comment (juhlee-microsoft) [inline Testscripts/Linux/SR-IOV-Utils.sh:80]: I think this can simpler than those lines. can you also use regex or something similar to the list 84?

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (juhlee-microsoft) [inline Testscripts/Linux/SRIOV-VerifyVF-Connection.sh:30]: This is the fixed version in RHEL and CentOS? 
what about other distro? 
is it hit this line?

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (juhlee-microsoft) [inline Testscripts/Linux/SRIOV-VerifyVF-Connection.sh]: check only two interfaces?

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (juhlee-microsoft) [inline Testscripts/Linux/SRIOV-VerifyVF-Connection.sh]: after this line, it does not check again. Need to redesign the step please.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/SR-IOV-Utils.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SR-IOV-Utils.sh:80]: Changed to one line command

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SRIOV-VerifyVF-Connection.sh:30]: Added a distro check

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SRIOV-VerifyVF-Connection.sh]: Added a recheck condition once after running dhclient

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SRIOV-VerifyVF-Connection.sh]: changed the logic to verify whether all the Extra NICs are having IP

**Comment by sireeshapotnuru** [comment]
Addressed two more Testcases in this PR

[LISAv2 Test Results Summary]
Test Run On           : 05/03/2019 08:01:05
ARM Image Under Test  : RedHat : RHEL : 6.10 : latest
Total Test Cases      : 2 (2 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:31

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-DISABLEVF-ON-GUEST                                                          PASS                 2.48 
    2 SRIOV                SRIOV-RELOAD-MODULE                                                               PASS                 2.36 

[LISAv2 Test Results Summary]
Test Run On           : 05/03/2019 08:01:44
ARM Image Under Test  : OpenLogic : CentOS : 7.5 : latest
Total Test Cases      : 2 (2 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:21

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 SRIOV                SRIOV-DISABLEVF-ON-GUEST                                                          PASS                 2.22 
    2 SRIOV                SRIOV-RELOAD-MODULE                                                               PASS                 2.22

File: Testscripts/Linux/SR-IOV-Utils.sh
Comment (lizzha) [inline Testscripts/Linux/SR-IOV-Utils.sh]: could you double check this on CentOS 7.6? On CentOS 7.6, the DISTRO_VERSION is 7.6.1810, which also matches this condition

**Review by lizzha — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (juhlee-microsoft) [reply Testscripts/Linux/SRIOV-VerifyVF-Connection.sh:30]: if $DISTRO_NAME is sles or ubuntu, it jumps to false statement, where is none. it does not skip.

File: Testscripts/Linux/SR-IOV-Utils.sh
Comment (juhlee-microsoft) [reply Testscripts/Linux/SR-IOV-Utils.sh:80]: I meant regex or something better approach not concise the commands in single line.

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (juhlee-microsoft) [inline Testscripts/Linux/SRIOV-VerifyVF-Connection.sh:40]: This logic is not so clear. This script is called in single VF as well as Max VF. In case of max VF, this block is not exercised.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/SR-IOV-Utils.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SR-IOV-Utils.sh]: Added a ^ to check whether the DISTRO_VERSION is starting with 6 series

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SRIOV-VerifyVF-Connection.sh:40]: Added a check if [[ "$NIC_COUNT" -gt 1 ]];then 
in case of single VF, NIC_COUNT=1 , this block will not get executed because we are always getting IP
in case of Max VF, NIC_COUNT >1 , this block will get executed and checks whether any of the NIC is not having IP and runs dhclient for that NIC

File: Testscripts/Linux/SRIOV-VerifyVF-Connection.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SRIOV-VerifyVF-Connection.sh:30]: In current implementation of LIS we are supporting SRIOV on centos, rhel, oracle 6.7 to 6.10 
SRIOV is not supported in 6.5 and 6.6 of centos, rhel, oracle.
Hence, this if blocked is added to check and skip if distro_version is < 6.7
for other distros there is no change in the test.

**Comment by sireeshapotnuru** [comment]
@lizzha , @juhlee-microsoft ,@adriansuhov please approve this PR and merge it

**Comment by juhlee-microsoft** [comment]
> @lizzha , @juhlee-microsoft ,@adriansuhov please approve this PR and merge it

@sireeshapotnuru  I commented back again. Please do not close unless I agree. Also please review my comment. I don't think I can approve unless you propose new design.

**Review by adriansuhov — APPROVED** [review]

**Review by lizzha — APPROVED** [review]

File: Testscripts/Linux/SR-IOV-Utils.sh
Comment (sireeshapotnuru) [reply Testscripts/Linux/SR-IOV-Utils.sh:80]: Taking reference from existing code in script - 
vf_interface=$(find /sys/devices/* -name "*${synthetic_interface}" | grep "pci" | sed 's/\// /g' | awk '{print $12}')

I have added the way which is widely used in the LISAv2 with grep and awk.
In line 84. sed is used to replace / with space

Comment (juhlee-microsoft) [reply Testscripts/Linux/SR-IOV-Utils.sh:80]: This is widely used in LISAv2? ok, let me check out later.

**Review by iamshital — APPROVED** [review]

**Review by juhlee-microsoft — APPROVED** [review]

**Pull request merged**
